transaction.go

```diff
@@ -33,6 +33,10 @@ var (
 	ErrInputOutputManaMismatch = ierrors.New("inputs and outputs do not contain the same amount of Mana")
 	// ErrInputCreationAfterTxCreation gets returned if an input has creation time after the transaction creation time.
 	ErrInputCreationAfterTxCreation = ierrors.New("input creation time after tx creation time")
+	// ErrUnknownTransactionEssenceType gets returned for unknown transaction essence types.
+	ErrUnknownTransactionEssenceType = ierrors.New("unknown transaction essence type")
+	//ErrUnexpectedUnderlyingTransactionType gets returned for unknown input type of transaction.
```

Comment:
the comment, the name and the errors message are all somehow different.
And the comment should have a whitespace at the beginning.